### PR TITLE
Show new lon/lat geocoding in background importer

### LINKED ISSUE #4381: Change lon/lat georeferencing to use the background importer
Extracted from #4262

> @josecruz wrote:
instead of this:
screen shot 2015-07-07 at 6 30 25 pm
- Geocoding status:
We should use this:
screen shot 2015-07-07 at 4 10 54 pm
* Regarding this format:
- The copy should be: Geocoding datasets...
- Should be the same icon as the geocoding modal one

--- 
(edit: added TODOs)

tasks:
- [x] remove current loading indicator at the top
- [x] display geocoding progress in the background importer
- [x] remove cancel-button since can't abort a lon/lat geocoding
- when background importer only have geocoding processes should
  - [x] show "geocoding dataset(s)…" instead
  - [x] show the same icon as for georef modal

### PR DESCRIPTION
**Pull request by viddo**
Fixes #4381 

screen shot 2015-07-17 at 15 34 25

screen shot 2015-07-17 at 15 34 02

@xavijam I think you're the best candidate to code review since you implemented the background importing
cc @josecruz @saleiva for visuals

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5096/))

File: lib/assets/javascripts/cartodb/common/background_polling/models/lon_lat_geocoding_model.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  hasCompleted: function() {
+    return this.get('hasCompleted');
+  },
+
+  hasFailed: function() {
+    return this.get('hasFailed');
+  },
+
+  getError: function() {
+    // TODO: correct?
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/common/background_polling/models/lon_lat_geocoding_model.js]: Whatever you decide is ok. I'm trying to "standarize" validationError local variable inside de model as new Backbone versions do, but having this ```getError``` public function is ok.

Comment (viddo) [inline lib/assets/javascripts/cartodb/common/background_polling/models/lon_lat_geocoding_model.js]: I'm just using the methods that already exists (just like the other models), the comment could be removed though

**Comment by xavijam** [comment]
@viddo be careful with where geocoding dialog is launched because it happens in several parts of the code. In the options menu (what you covered marvelous), in the geo label that appear close to the the_geom column title and in the old dialog that it suggested there wasn't geo data in your dataset.

File: lib/assets/javascripts/cartodb/common/background_polling/models/lon_lat_geocoding_model.js (hunk 1 above)
Comment (viddo) [inline lib/assets/javascripts/cartodb/common/background_polling/models/lon_lat_geocoding_model.js]: *using=defining the methods expected to exist on the model (based on looking at the other ones)

**Comment by viddo** [comment]
Ah oh, I missed those. :( Thx for pointing that out, I'll get to it

**Comment by xavijam** [comment]
No problem, apart from that, if it is working properly, ready to go :).

File: lib/assets/javascripts/cartodb/table/table.js [hunk 2]
```diff
@@ -355,6 +355,57 @@ $(function() {
       this.workView.active(this.workViewActive);
     },
 
+    _initBinds: function() {
+      cdb.god.bind('geocodingChosen', this._onGeocodingChosen, this);
```
Comment (viddo) [inline lib/assets/javascripts/cartodb/table/table.js]: @xavijam Yes, I know, we should avoid this usage,, but I didn't see any other way and I'll explain way: some sub-views to table are dropdowns, [which are cleaned](https://github.com/CartoDB/cartodb/blob/master/lib/assets/javascripts/cartodb/old_common/header.js#L111-L129) once the georeference modal is opened, so there's no way (AFAIK reasonable w/o worse hacks) to listen to a re-triggered event on a non-existing view. In this case I think it's justified to use the cdb.god model, unless you have some other proposal?

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5116/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5133/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5150/))

**Comment by josecruz** [comment]
Yeah :+1:

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5163/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5167/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5169/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5174/))

**Pull request merged**